### DISCUSSION TITLE
Vite: Disable bundling in build?

### DISCUSSION THREAD

**Discussion by fregante** [crxjs/chrome-extension-tools#804] [Q&A]
I'm looking for a solution that lets me ship "raw" source files without bundling them. I saw that this Vite plugin does exactly what I'm looking for _during development:_

- it bundles each vendor from node_modules separately (`dist/vendor/deps/*.js`)
- it generates 1 file for each 1 source file, even preserving their original directory structure

But once I run the final `build` command, I get:

- a single content.js file with everything in it

Is it possible to preserve the 1:1 file structure and disable bundling in the `build` step?

---

Context: I followed this guide with Vite 2, TypeScript: https://crxjs.dev/vite-plugin/getting-started/vanilla-js/create-project

**Comment by guypassy** [comment]
You can use [rollupOptions's manualChunks](https://rollupjs.org/configuration-options/#output-manualchunks) to manage what dependencies do not get bundled, or bundled separately.

```js
//vite.config.js
build: {
  rollupOptions: {
    output: {
      manualChunks: {
        lodash: ['lodash']
      }
    }
  }
}
```

If you want each dependency to be bundled separately, here's a blog with a nice solution.
https://sambitsahoo.com/blog/vite-code-splitting-that-works.html
Basically, import the list of dependencies from `./package.json` and iterate on it to add separate chunks.
